recipes/perl/assets/create_assetGroup.pl

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env perl
+
+use LWP::UserAgent;
+use LWP::Protocol::https;
+print "LWP::UserAgent: ".LWP::UserAgent->VERSION,"\n";
+print "LWP::Protocol::https: ".LWP::Protocol::https->VERSION,"\n";
+use JSON;
+use Getopt::Long qw(GetOptions);
+
+
+#
+# The token is the key to the NetBackup AuthN/AuthZ scheme.  You must login and get a token
+# and use this token in your Authorization header for all subsequent requests.  Token validity
+# is fixed at 24 hours
+#
+
+my $content_type = "application/vnd.netbackup+json; version=4.0";
+my $protocol = "https";
+my $port = "1556";
+my $nbmaster;
+my $username;
+my $password;
+my $domainName;
+my $domainType;
+my $token;
+my $base_url;
+
+my $ua = LWP::UserAgent->new(
+                ssl_opts => { verify_hostname => 0, verify_peer => 0},
+            );
+
+# subroutines for printing usage and library information required to run the script.
+sub print_usage {
+    print("\n\nUsage:");
+    print("\nperl get_vmware_assets -nbmaster <masterServer> -username <username> -password <password> [-domainName <domainName>] [-domainType <domainType>] \n\n\n");
+}
+
+sub print_disclaimer {
+    print("--------------------------------------------------------\n");
+    print("--       This script requires Perl 5.20.2 or later    --\n");
+    print("--------------------------------------------------------\n");
+    print("Executing this library requires some additional libraries like \n\t'LWP' \n\t'JSON'\ \n\t'Getopt'\ \n\n");
+    print("You can specify the 'nbmaster', 'username', 'password', 'domainName', 'domainType' as command-line parameters\n");
+    print_usage();
+}
+
+
+# subroutine to process user input
+sub user_input {
+    GetOptions(
+        'nbmaster=s' => \$nbmaster,
+        'username=s' => \$username,
+        'password=s' => \$password,
+        'domainName=s' => \$domainName,
+        'domainType=s' => \$domainType,
+    ) or die print_usage();
```

Comment:
This will print the usage message to stdout, then print "1" and the current source location to stderr because `print_usage` returns the result of its last statement, which is `print`, and `die` prints its argument. The usual failed method goes `GetOptions(...) or pod2usage`, but if you don't want to bring in more modules (even core ones), then maybe this instead:

```suggestion
    ) or (print_usage() && exit);
```

That would match the behavior of the error checking below.

Reply:
ok.will make the change.